aws_lambda_powertools/utilities/parameters/appconfig.py

```diff
@@ -84,7 +84,7 @@ def __init__(
 
         super().__init__()
 
-    def _get(self, name: str, **sdk_options) -> str:
+    def _get(self, name: str, **sdk_options) -> bytes:
```

Comment:
This makes MyPy fail in a different way, since this overrides an abstract method return type previously defined as `str` in `BaseProvider` -- I'm gonna add to the papercut list to ensure we run MyPy checks as part of CI; it's gonna increase the bar for contribution but reduce these type of issues as they seem to be growing elsewhere too.

`error: Return type "bytes" of "_get" incompatible with return type "str" in supertype BaseProvider`

There's also another problem - `get_app_config()` returns `Union[str, list, dict]` when it's in fact `bytes`, unless Transform is used. 

A quick non-breaking change fix for now might be having the base `_get` and `get_app_config` to accept multiple return types.

A proper solution would involve [`@overload` type when using specific transform](https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading) (json, binary), and an explicit cast to the expected type in certain areas (needs evaluation w/ MyPy and tests).

Reply:
summary comment updated to reflect and remove the ambiguous part now that we have a better understanding